xtevent/_eventols.ado

```diff
@@ -159,17 +170,98 @@ program define _eventols, rclass
 		}
 	}
 	else loc indepvars ""
+		
+	******** SA *******
+	if "`sun_abraham'"!=""{
+		* Parse the dependent variable
+		local lhs = "`depenvar'"
+		local rel_time_list = "`included'"
+		
+		* Convert the varlist of relative time indicators to nvarlist 
+		local nvarlist "" // copies of relative time indicators with control cohort set to zero
+		local dvarlist "" // for display
+		foreach l of varlist `rel_time_list' {
+			local dvarlist "`dvarlist' `l'"
+			tempname n`l'
+			qui gen `n`l'' = `l'
+			qui replace `n`l'' = 0 if  `control_cohort' == 1 
+			local nvarlist "`nvarlist' `n`l''" 
+		}	
+
+		* Get cohort count  and count of relative time
+		qui levelsof `cohort' if  `control_cohort' == 0, local(cohort_list) 
+		local nrel_times: word count `nvarlist' 
+		local ncohort: word count `cohort_list'  
+		
+		**** step 2 
+		* Initiate empty matrix for weights 
+		tempname bb ff_w
+		
+		* Loop over cohort and get cohort shares for relative times
+		local nresidlist ""
+		foreach yy of local cohort_list {
+			tempvar cohort_ind resid`yy'
+			qui gen `cohort_ind'  = (`cohort' == `yy') 
+			qui regress `cohort_ind' `nvarlist'  if `touse' & `control_cohort' == 0 [`weight'`exp']  , nocons

```

Comment:
@Constantino-Carreto-Romero Check if you do this with `_regress`. It's slightly faster.